apps/web/src/app/(private)/settings/_components/ProviderApiKeys/New/index.tsx

```diff
@@ -51,44 +55,81 @@ export default function NewApiKey({
     >
       <form id='createApiKeyForm' action={action}>
         <FormWrapper>
-          <FormField label='Provider'>
-            <Select
-              required
-              name='provider'
-              onChange={(newValue) => setProviderType(newValue)}
-              defaultValue={data?.provider}
-              options={PROVIDER_OPTIONS}
-            />
-          </FormField>
-          <FormField label='Name'>
+          <FormField
+            label='Provider'
+            description="This is the name you'll use in the prompt editor to refer to use this provider and model."
+          >
             <Input
               required
               type='text'
               name='name'
-              defaultValue={data?.name}
-              placeholder='My API Key'
+              defaultValue={result?.name}
+              placeholder='My Provider'
+              // @ts-expect-error
+              errors={errors?.name}
```

Comment:
Why this? Let's find out

Reply:
I am assuming this `errors` prop allows to show an specific error underneath the field if there is any after submitting the form. I saw this pattern in others forms such as `apps/web/src/app/(public)/setup/SetupForm/index.ts`... so I though it was missing and just added this for consistency.

I see other forms that do `const errors = error?.fieldErrors as Record<string, string[]>` to avoid putting a ` // @ts-expect-error` repeatedly. Will change for that or I can just remove the errors field entirely if not necessary.

Reply:
The question is why `error.fieldErrors` is not typed like this. What type is expecting the `errors` prop? Don't worry we can check on Monday